includes/class-wc-stripe-apple-pay-registration.php

```diff
@@ -241,34 +242,45 @@ public function process_apple_pay_verification() {
 	 * @since 4.0.6
 	 */
 	public function admin_notices() {
-		if ( ! $this->stripe_enabled ) {
+		if ( ! $this->stripe_enabled || ! $this->payment_request ) {
 			return;
 		}
 
 		if ( ! current_user_can( 'manage_woocommerce' ) ) {
 			return;
 		}
 
-		if ( $this->payment_request && ! empty( $this->apple_pay_verify_notice ) ) {
-			$allowed_html = array(
-				'a' => array(
-					'href'  => array(),
-					'title' => array(),
-				),
-			);
-
-			echo '<div class="error stripe-apple-pay-message"><p>' . wp_kses( make_clickable( $this->apple_pay_verify_notice ), $allowed_html ) . '</p></div>';
+		$empty_notice = empty( $this->apple_pay_verify_notice );
+		if ( $empty_notice && ( $this->apple_pay_domain_set || empty( $this->secret_key ) ) ) {
+			return;
 		}
 
 		/**
 		 * Apple pay is enabled by default and domain verification initializes
 		 * when setting screen is displayed. So if domain verification is not set,
 		 * something went wrong so lets notify user.
 		 */
-		if ( ! empty( $this->secret_key ) && $this->payment_request && ! $this->apple_pay_domain_set ) {
+		$allowed_html = array(
+			'a' => array(
+				'href'  => array(),
+				'title' => array(),
+			),
+		);
+
+		echo '<div class="error stripe-apple-pay-message"><p>';
+
+		echo 'Apple Pay domain verification failed';
+		$reason = $empty_notice ? '' : ' with the error: ' . wp_kses( make_clickable( $this->apple_pay_verify_notice ), $allowed_html ) . '';
```

Comment:
I like the concise way these cases are combined here, but it's at the cost of the translated string – is that OK? We could just have two separate full translated sentences here, or skip "with the error". I don't know if Stripe's error message would ever be localized, so it might not be that big a deal.

Reply:
I assume you're referring to the `'with the error:'` string? If that's the case are you opposed to how I fixed that in 456e0e22e880a07305e621e3d330030e5da286dc?

If you're referring to the Stripe error, that's already sort of out of our control. Maybe it'd be better to simply skip showing that error and instead always direct people to the log? Doing that would certainly simplify the logic a bit here.

What do you think @allendav?

Reply:
Ah, not Stripe's error, which is unchanged here – I was just referring to "Apple Pay domain verification failed" and "Apple Pay domain verification failed with the error:" – which I'd probably consider two separate full strings for purposes of translation, though I'm not sure which languages would be affected in practice – AFAIK the way it's done in https://github.com/woocommerce/woocommerce-gateway-stripe/commit/456e0e22e880a07305e621e3d330030e5da286dc may be sufficient.

Reply:
Actually, thinking about this more, I think you're right; this probably should be 2 separate strings. I think it might be hard for translators to understand the context of "with the error". I'll make this 2 separate strings and use `$empty_notice` to decide which one to use.

Reply:
I pushed an update in 5b536e3aaf158df19ac068c2389593c0e72a075f, what do you think?

Also, should I update the pot file in this PR, or will it be updated during the release?

Reply:
Looks good to me – not sure about the POT for this PR, but I _think_ it is (or should be) a release task.

Reply:
The POT will be regenerated by `npm run build` when we release it. PRs can do it too, but its ok to let it wait for release.